### ISSUE TITLE
Bad User Experience

### ISSUE THREAD

**Issue by fneu** [coala/coala-bears#512]
hey guys,
### tl;dr

it would be really cool to have a compact list that shows all available bears and their dependencies / + whether they are met.
### story

so I'm in the process of creating a docker image for coala right now, containing all the stuff necessary to run everything we can offer. I'm not done and bear stuff breaks. One thing I notice is that it is rather easy to figure out missing or broken dependencies from running the tests pulled from github. But those don't come with coala by default so I'm afraid we are offering a really bad user experience for others.

Say I'm a user looking to check R stuff. I install coala and coala-bears (and in this case also some R related stuff I found in `.ci/deps.sh`on GitHub)

I'm now checking for bears I can use on my R files, first result is:

```
59c91643698e:/coala-bears # coala -l R
FormatRBear:
  This bear checks and corrects formatting of R Code using known formatR utility.

  Supported languages:
   * R

  Used in:
   * linecounting
   * python
   * autopep8
   * todos
   * commit
   * default
   * linelength

  No needed settings.

  No optional settings.
```

So now I'm happy and want to use that bear

```
59c91643698e:/coala-bears # coala -b FormatRBear
Executing section Default...
[WARNING][10:49:43] Bear FormatRBear failed to run. Take a look at debug messages for further information.
[WARNING][10:49:43] Bear FormatRBear failed to run. Take a look at debug messages for further information.
[WARNING][10:49:43] Bear FormatRBear failed to run. Take a look at debug messages for further information.
[WARNING][10:49:43] Bear FormatRBear failed to run. Take a look at debug messages for further information.
[WARNING][10:49:43] Bear FormatRBear failed to run. Take a look at debug messages for further information.
[WARNING][10:49:43] Bear FormatRBear failed to run. Take a look at debug messages for further information.
[WARNING][10:49:43] Bear FormatRBear failed to run. Take a look at debug messages for further information.
[WARNING][10:49:43] Bear FormatRBear failed to run. Take a look at debug messages for further information.
[WARNING][10:49:43] Bear FormatRBear failed to run. Take a look at debug messages for further information.
[WARNING][10:49:43] Bear FormatRBear failed to run. Take a look at debug messages for further information.
[WARNING][10:49:43] Bear FormatRBear failed to run. Take a look at debug messages for further information.
...
```

well I looked at the test and not debug message but still:

```
E           AssertionError: Bear returned None on execution
E            
E           Running bear FormatRBear...
E           Bear FormatRBear failed to run. Take a look at debug messages for further information.
E           The bear FormatRBear raised an exception. If you are the writer of this bear, please make sure to catch all exceptions. If not and this error annoys you, you might want to get in contact with the writer of this bear.
E           
E           Traceback information is provided below:
E           
E           Traceback (most recent call last):
E             File "/usr/lib/python3.5/site-packages/coalib/bears/Bear.py", line 116, in execute
E               return list(self.run_bear_from_section(args, kwargs) or [])
E             File "/usr/lib/python3.5/site-packages/coalib/bears/Bear.py", line 109, in run_bear_from_section
E               return self.run(*args, **kwargs)
E             File "/coala-bears/bears/r/FormatRBear.py", line 24, in run
E               return self.lint(filename, file)
E             File "/usr/lib/python3.5/site-packages/coalib/bearlib/abstractions/Lint.py", line 102, in lint
E               results = self.process_output(results_output, filename, file)
E             File "/coala-bears/bears/r/FormatRBear.py", line 16, in process_output
E               output = output[:-1] + (output[-1].strip()+"\n",)
E           IndexError: tuple index out of range
E           Command: Rscript -e 'library(formatR)' -e 'formatR::tidy_source("/tmp/tmpx0rnz8xf")'
E           Stdout:
E           Stderr:
E            Error in library(formatR) : there is no package called ‘formatR’
E           Execution halted
```

there needs to be a better way

**Comment by AbdealiJK** [comment]
@Adrianzatreanu can this be moved to cib ?
Does cib have a `cib list` likefunction which can do it ? I really like `npm`'s tree like approach to show packages and their dependencies in a nice manner

**Comment by fneu** [comment]
Good call, cib is probably our best shot at improving this right now. IIRC cib has the `cib list` and `cib check-deps` functions, the latter of which Checks the dependency state of all "installed" bears.

The ui of cib is rather bad at this point, npm is better in that regard. That is better discussed there though.

Btw. Is it planned to somehow move cib into main-coala later on? Might be really useful soon.

**Comment by sils1297** [comment]
We hoped cib would be default for 0.9

**Comment by Adrianzatreanu** [comment]
The issue should be moved, indeed.

**Comment by fneu** [comment]
Code looks a Bit immature so far, entire Tool is one huge Funktion? Anyway 
I'm quite motivated to work some things on it.


Am 2. September 2016 10:13:18 nachm. schrieb Lasse Schuirmann 
<notifications@github.com>:

**Comment by jayvdb** [comment]
cib & package_manager are solving this.
